vtex-apps/telemarketing
# Add user creation if the user is not found

#### What problem is this solving?

When an agent telesales with a non-existent user, it just refreshes the page. There's a PR now that displays an error if a user doesn't exist (https://github.com/vtex-apps/telemarketing/pull/71), however, I think we can go beyond that and simply create the user and use it to login.  There's no reason why that shouldn't be a feature, considering that a user is automatically created during the checkout process when an order is placed (An user with just the email provided, no name, nothing else)

#### How to test it?

Make sure you have telemarketing configured and the appropriate role to your user, link the app and try to use an email that doesn't exist in the "CL" entity. It should create it and log you in with it.

#### Describe alternatives you've considered, if any.

No other alternatives.  This makes sense and makes the lives of our clients' agents alot easier.

#### How does this PR make you feel? [:link:](http://giphy.com/)

![](https://media.giphy.com/media/aMh59aKR8vjdC/giphy.gif)
